editor/src/core/commenting/comment-hooks.tsx

```diff
@@ -261,3 +261,79 @@ export function useUnresolvedThreads() {
     threads: threads.threads.filter((t) => t.metadata.resolved !== true),
   }
 }
+
+export function useSetThreadReadStatusOnMount(thread: ThreadData<ThreadMetadata> | null) {
+  const setThreadReadStatus = useSetThreadReadStatus()
+
+  React.useEffect(() => {
+    if (thread == null) {
+      return
+    }
+
+    setThreadReadStatus(thread.id, 'read')
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []) // only run it once on mount, because opening the popup means reading the thread, no dependencies added
+}
+
+export function useMyThreadReadStatus(thread: ThreadData<ThreadMetadata> | null): ThreadReadStatus {
+  const self = useSelf()
+  return useStorage((store) => {
+    if (thread == null) {
+      return 'unread'
+    }
+    const statusesForThread = store.userReadStatusesByThread[thread.id]
+    if (statusesForThread == null) {
+      return 'unread'
+    }
+    return statusesForThread[self.id] === true ? 'read' : 'unread'
+  })
+}
+
+export type ThreadReadStatus = 'read' | 'unread'
+
+export function useSetThreadReadStatus() {
```

Comment:
This should be named `useToggleThreadReadStatus`

Reply:
Why toggle? Toggle means switching to the opposite value, but this returns a function with which you can set to any given value.

Reply:
I think my reading comprehension has failed me today